sakai-evaluation-impl/src/java/org/sakaiproject/evaluation/logic/EvalEmailsLogicImpl.java

```diff
@@ -1279,27 +1282,39 @@ protected void logConsolidatedEmailRecipients(String emailType, String[] recipie
      * (non-Javadoc)
      * @see org.sakaiproject.evaluation.logic.EvalEmailsLogic#sendEvalSubmissionConfirmationEmail(java.lang.Long)
      */
-	public String sendEvalSubmissionConfirmationEmail(String userId, Long evaluationId) {
-	    String to = null;
-	    Boolean sendConfirmation = (Boolean) settings.get(EvalSettings.ENABLE_SUBMISSION_CONFIRMATION_EMAIL);
-
-	    if (sendConfirmation) {
-	        EvalEvaluation eval = getEvaluationOrFail(evaluationId);
-	        String from = getFromEmailOrFail(eval);
-	        //get the template
-	        EvalEmailTemplate emailTemplate = getEmailTemplateOrFail(EvalConstants.EMAIL_TEMPLATE_SUBMITTED, evaluationId);
-	        if (emailTemplate != null) {
-	            //make email and do the variable substitutions
-	            EvalEmailMessage em = makeEmailMessage(emailTemplate.getMessage(), emailTemplate.getSubject(), eval, null);
-	            // send the actual email for this user
-	            String[] emailAddresses = sendUsersEmails(from, new String[]{userId}, em.subject, em.message);
-	            if (emailAddresses.length > 0){
-	                LOG.info("Sent Submission Confirmation email to " + userId + ". (attempted to send to "+emailAddresses.length+")");	                
-	                commonLogic.registerEntityEvent(EVENT_EMAIL_SUBMISSION, EvalEvaluation.class, eval.getId().toString());
-	            }
-	        }
-	    }
-	    return to;
-	}
+    public String sendEvalSubmissionConfirmationEmail(String userId, Long evaluationId) {
+        String to = null;
+        Boolean sendConfirmation = (Boolean) settings.get(EvalSettings.ENABLE_SUBMISSION_CONFIRMATION_EMAIL);
+
+        if (sendConfirmation) {
+            EvalEvaluation eval = getEvaluationOrFail(evaluationId);
+            String from = getFromEmailOrFail(eval);
+            //get the template
+            EvalEmailTemplate emailTemplate = getEmailTemplateOrFail(EvalConstants.EMAIL_TEMPLATE_SUBMITTED, evaluationId);
+            if (emailTemplate != null) {
+
+                // Subject and message should never be empty or null - EVALSYS-1441
+                String subject = emailTemplate.getSubject();
+                if (!StringUtils.isNotBlank(subject)) {
```

Comment:
Mea culpa on the original code here, but seems this could more concisely be just 

if (StringUtils.isBlank(subject)) {

and the same below for message.